extensions/apicurio-registry-avro/runtime/src/main/java/io/quarkus/apicurio/registry/binding/ServiceRegistryBindingConverter.java

```diff
@@ -0,0 +1,107 @@
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.ConfigProvider;
+
+import io.quarkus.kubernetes.service.binding.runtime.ServiceBinding;
+import io.quarkus.kubernetes.service.binding.runtime.ServiceBindingConfigSource;
+import io.quarkus.kubernetes.service.binding.runtime.ServiceBindingConverter;
+
+public class ServiceRegistryBindingConverter implements ServiceBindingConverter {
+
+    private static Logger LOG = Logger.getLogger(ServiceRegistryBindingConverter.class.getName());
+
+    private static final String INCOMING_PREFIX = "mp.messaging.incoming.";
+    private static final String OUTGOING_PREFIX = "mp.messaging.outgoing.";
+
+    @Override
+    public Optional<ServiceBindingConfigSource> convert(List<ServiceBinding> serviceBindings) {
+        var matchingByType = ServiceBinding.singleMatchingByType("serviceregistry", serviceBindings);
+        Config config = ConfigProvider.getConfig();
+        if (matchingByType.isEmpty()) {
+            return Optional.empty();
+        }
+
+        var binding = matchingByType.get();
+
+        List<String> channels = extractChannels(config);
+
+        Map<String, String> properties = new HashMap<>();
+
+        String registryUrl = binding.getProperties().get("registryUrl");
+        if (registryUrl == null) {
+            registryUrl = binding.getProperties().get("registryurl");
+        }
+        if (registryUrl != null) {
+            properties.put("mp.messaging.connector.smallrye-kafka.apicurio.registry.url", registryUrl);
+        }
+        
+
+        for (String channel : channels) {
+
+            String prefix = channel;
+
+            String oAuthHost = binding.getProperties().get("oauthServerUrl");
+            if (oAuthHost == null) {
+                oAuthHost = binding.getProperties().get("oauthserverurl");
+            }
+            if (oAuthHost != null) {
+                properties.put(prefix + "apicurio.auth.service.url", oAuthHost);
+            }
+
+            String clientId = binding.getProperties().get("clientId");
+            if (clientId == null) {
+                clientId = binding.getProperties().get("clientid");
+            }
+            if (clientId != null) {
+                properties.put(prefix + "apicurio.auth.client.id", clientId);
+            }
+
+            String clientSecret = binding.getProperties().get("clientSecret");
+            if (clientSecret == null) {
+                clientSecret = binding.getProperties().get("clientsecret");
+            }
+            if (clientSecret != null) {
+                properties.put(prefix + "apicurio.auth.client.secret", clientSecret);
+            }
+
+            String realm = binding.getProperties().get("oauthRealm");
+            if (realm == null) {
+                realm = binding.getProperties().get("oauthRealm");
+            }
+            if (clientSecret != null) {
+                properties.put(prefix + "apicurio.auth.realm", realm);
```

Comment:
[Question] Should user properties take precedence over binding ones?

Reply:
I don't think so. You wouldn't want a property in an example project or quickstart to overwrite something that a user had done. I can see a situation where we have dummy properties setup so the application starts, but expect them to be replaced.